### PR TITLE
first run through commit

### PR DESCRIPTION
**Pull request by madisonmay**

### CODE REVIEW

File: enso/experiment/crfac.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+        self.param_grid = {
+            "penalty": ["l2"],
+            "max_iter": [500],
+            "C": [0.1, 1.0, 10.0, 100.0, 1000.0],
+            "solver": ["lbfgs"],
+            "multi_class": ["multinomial"],
+        }
+    def spacy_vectorize(self, texts):
+        return [i.vector for i in self.NLP.pipe(texts, disable=['parser', 'ner', 'textcat'])]
```
Comment (madisonmay) [inline enso/experiment/crfac.py]: Some of these vectors may be all zeros -- might be fine just an FYI.

File: enso/metrics/basic_rationalized.py [hunk 2]
```diff
@@ -12,14 +13,18 @@ class MacroRocAucRationalized(ClassificationMetric):
...
+        print("DEBUGING INFO HERE!!!!")
+        print(Counter(ground_truth))
         classes = list(set(ground_truth) | set(result.columns))
         classwise_auc = {}
         binary_labels = labels_to_binary(ground_truth)
         default = np.zeros(shape=[len(ground_truth), 1], dtype=np.int32)
         binary_labels = np.hstack([binary_labels[column].values.reshape(-1, 1) if column in binary_labels else default for column in classes])
-        predicted_labels = np.hstack([result[column].values.reshape(-1, 1) if column in result.columns else default for column in classes])
-        return roc_auc_score(binary_labels, predicted_labels, average='macro')
+        predicted_labels = np.hstack([result[column].values.reshape(-1, 1) if column in result.columns else default for column in classes]) return roc_auc_score(binary_labels, predicted_labels, average='macro')
```
Comment (madisonmay) [inline enso/metrics/basic_rationalized.py]: This line looks like a syntax error to me?

File: enso/experiment/crfac.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+"""Module for any LR-style experiment."""
+from sklearn.linear_model import LogisticRegression
+import sklearn_crfsuite
+import numpy as np
+import pandas as pd
+import spacy
+import crf_processing
```
Comment (madisonmay) [inline enso/experiment/crfac.py]: Doesn't seem like you committed this file

File: enso/experiment/crfac.py (hunk 1 above)
Comment (Concomitant) [reply enso/experiment/crfac.py]: Yup! I'm counting on that. Seems to be the behavior I'm getting.

File: enso/metrics/basic_rationalized.py (hunk 2 above)
Comment (Concomitant) [reply enso/metrics/basic_rationalized.py]: yeah, fixed that one.

File: enso/experiment/crfac.py (hunk 3 above)
Comment (Concomitant) [reply enso/experiment/crfac.py]: just pushed it. Yeah, I also probably want to factor that out in some better way.

File: enso/experiment/crfac.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        }
+    def spacy_vectorize(self, texts):
+        return [i.vector for i in self.NLP.pipe(texts, disable=['parser', 'ner', 'textcat'])]
+
+
+    def fit(self, X, Y):
+
+        # Make special features for the CRF model
+        spans = crf_processing.get_spans_enso(X, Y)
+        spantexts = [crf_processing.get_span_texts(span) for span in spans]
```
Comment (benleetownsend) [inline enso/experiment/crfac.py]: (just making notes to follow the code through) span text is a list of tuples `(text, "0" or "rationale")`

File: enso/experiment/crfac.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        features = [crf_processing.make_crf_features(spantext) for spantext in spantexts]
```
Comment (benleetownsend) [inline enso/experiment/crfac.py]: features is a list of 3-tuples (span_text if o, else token text, "WORD", BIO encoded rationale")

File: crf_processing.py [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+    tokens = [(t.text, 'WORD') for t in tokenizer(text)]
+    crf_feats = sent2features(tokens)
+    crf_preds = crf.predict_single(crf_feats)
+    text_out = ' '.join([tok[0] for tok, pred in zip(tokens, crf_preds) if pred != 'O'])
+    return text_out
+
+
+def word2features(sent, i):
+    word = sent[i][0]
+    postag = sent[i][1]
```
Comment (benleetownsend) [inline crf_processing.py]: My understanding is that this will always be just "WORD"?

File: enso/metrics/basic_rationalized.py [hunk 7]
```diff
@@ -2,6 +2,7 @@
 from sklearn.metrics import roc_auc_score, accuracy_score, log_loss
 import numpy as np
 
+from collectiosn import Counter
```
Comment (madisonmay) [inline enso/metrics/basic_rationalized.py]: syntax error again

Comment (Concomitant) [reply enso/metrics/basic_rationalized.py]: Got it

File: enso/experiment/__init__.py [hunk 8]
```diff
@@ -327,11 +332,17 @@ def _split_dataset(dataset, training_size):
...
+    def _load_dataset(dataset_name, featurizer_name, min_lf=0):
         """Responsible for loading a given dataset given the dataset_name and featurizer."""
         read_location = feature_set_location(dataset_name, featurizer_name)
         logging.info("Loading Dataset: %s" % read_location)
-        return joblib.load(read_location)
+        if min_lf == 0:
+            return joblib.load(read_location)
+        unfiltered = joblib.load(read_location)
+        label_counts = Counter([j[1] for j in unfiltered.Targets])
+        drop_labels = [i for i in label_counts if label_counts[i] >= min_lf]
```
Comment (madisonmay) [inline enso/experiment/__init__.py]: Confusing name for this variable

Comment (madisonmay) [reply enso/experiment/__init__.py]: Should also change to handle more than just rationalized classification before merge

**Pull request closed without merging**